lib/puppet/resource/type.rb

```diff
@@ -64,49 +59,6 @@ class Puppet::Resource::Type
     define_method("#{t}?") { self.type == t }
   end
 
-  require 'puppet/indirector'
-  extend Puppet::Indirector
-  indirects :resource_type, :terminus_class => :parser
-
-  def self.from_data_hash(data)
```

Comment:
@joshcooper had said it was okay to remove these because they would only be called in the context of the resource_types indirection, right (and not in the context of serialization for any other indirection, like for a catalog response)?